contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -190,20 +161,34 @@ contract LPOptimiserStrategy is DefaultAccessControl {
             PRBMathSD59x18.log2(1000100000000000000)
         );
 
+        // 5. Scale ticks from wad
         int256 newTickLower = newTickLowerWad / 1e18;
         int256 newTickUpper = newTickUpperWad / 1e18;
 
-        newTickLowerMul = nearestTickMultiple(int24(newTickLower), _tickSpacing);
-        newTickUpperMul = nearestTickMultiple(int24(newTickUpper), _tickSpacing);
-
-        // Update the global upper and lower tick variables
-        _tickLower = newTickLowerMul;
-        _tickUpper = newTickUpperMul;
+        // 6. The underlying Voltz VAMM accepts only ticks multiple of tickSpacing
+        // Hence, we get the nearest usable tick
+        newTickLowerMul = nearestTickMultiple(int24(newTickLower), tickSpacing);
+        newTickUpperMul = nearestTickMultiple(int24(newTickUpper), tickSpacing);
 
         // Call to VoltzVault contract to update the position lower and upper ticks
-        // _vault.rebalance(IVoltzVault.TickRange(_tickLower, _tickUpper));
+        _vault.rebalance(IVoltzVault.TickRange(newTickLowerMul, newTickUpperMul));
 
-        emit Rebalanced(newTickLowerMul, newTickUpperMul);
+        emit RebalancedTicks(newTickLowerMul, newTickUpperMul);
         return (newTickLowerMul, newTickUpperMul);
     }
+
+    /// @notice Callback function called after for ERC20RootVault::deposit
+    function depositCallback() external override {
```

Comment:
interesting, why does the depositCallback need to be in the strategy, doesn't seem that intuitive to me architecturally @Sol3iman @0xZenus

Reply:
The idea is as follows:
   - ERC20 Root Vault has sub-vaults and the underlying strategy (e.g. For Voltz: the subvaults are **erc20Vault** and **voltzVault** and the strategy is **LPOptimiserStrategy** -- the first subvault must be **erc20Vault**)
   - users deposit funds into ERC20 Root Vault and ERC20 Root Vault automatically transfers these funds to **erc20Vault** and then calls this callback. 
   - the callback is in strategy because that's the piece that dictates how funds are handled in the subvaults (i.e. For Voltz: the callback pulls all funds from **erc20Vault** to **voltzVault**)